__tests__/produce.ts

```diff
@@ -70,27 +71,47 @@ it("can update readonly state via standard api", () => {
 // NOTE: only when the function type is inferred
 it("can infer state type from default state", () => {
     type State = {readonly a:number} | boolean
-    type Recipe = (base?: State | undefined) => State
+    type Recipe = (base?: State | boolean) => State
 
-    let foo = produce(_ => {}, {} as State)
+    let foo = produce((x: any) => {}, {} as State)
     exactType(foo, {} as Recipe)
 })
 
 it("can infer state type from recipe function", () => {
     type State = {readonly a: string} | {readonly b: string}
-    type Recipe = (base: State | undefined, arg: number) => State
+    type Recipe = (base: State) => State
 
-    let foo = produce((draft: Draft<State>, arg: number) => {}, {} as any)
+    let foo = produce((draft: Draft<State>) => {})
+    exactType(foo, {} as Recipe)
+})
+
+it("can infer state type from recipe function with arguments", () => {
+    type State = {readonly a: string} | {readonly b: string}
+    type Recipe = (base: State, x: number) => State
+
+    let foo = produce((draft: Draft<State>, x: number) => {})
+    exactType(foo, {} as Recipe)
+})
+
+it("can infer state type from recipe function with arguments and initial state", () => {
+    type State = {readonly a: string} | {readonly b: string}
+    type Recipe = (base: State | undefined, x: number) => State
+
+    let foo = produce((draft: Draft<State>, x: number) => {}, {} as State)
     exactType(foo, {} as Recipe)
 })
 
 it("cannot infer state type when the function type and default state are missing", () => {
-    const res = produce(_ => {})
+    const res = produce((_: any) => {})
     exactType(res, {} as (base: any) => any)
+
+    // slightly different type inference...
+    const res2 = produce((_) => {})
```

Comment:
this case is a bit weird, but `(_: any) => ` is seen by TS as a single arg function, while `_ =>` is variadic...